src/shader.cpp

```diff
@@ -233,29 +244,37 @@ bool ShaderProgram::setShaderFromSourceFile(QString fileName)
 {
     QFile file(fileName);
     if (!file.open(QFile::ReadOnly))
+    {
+        //g_logger.error("Could not read Shader:\n%s", fileName.toStdString());
         return false;
+    }
     return setShader(file.readAll());
 }
 
 
 bool ShaderProgram::setShader(QString src)
 {
-    std::unique_ptr<Shader> vertexShader(new Shader(QGLShader::Fragment, m_context));
-    std::unique_ptr<Shader> fragmentShader(new Shader(QGLShader::Vertex, m_context));
+    std::unique_ptr<Shader> vertexShader(new Shader(QGLShader::Fragment));
+    std::unique_ptr<Shader> fragmentShader(new Shader(QGLShader::Vertex));
     //tfm::printf("Shader source:\n###\n%s\n###\n", src.toStdString());
-    if(!vertexShader->compileSourceCode(src.toAscii()))
+#ifdef DISPLAZ_USE_QT4
+    QByteArray src_ba = src.toAscii();
+#else
+    QByteArray src_ba = src.toLatin1();
+#endif
```

Comment:
In my experience the actual source code isn't the problem when in comes to weird characters, but the comments can be. Sometimes weird characters can end up inside comments, but I guess `toUtf8()` would handle these cases for us as well.